### PR TITLE
Introduce zendesk client with sanitations

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
$subject. 

Related to https://github.com/ballerina-platform/ballerina-library/issues/6185

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: docs/spec/sanitations.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    * Therefore, the `--nullable` option is used as a precaution to avoid potential data-binding issues in the runtime,  
+      which will generate all the request/response type fields with the support to handle null values.
+    * This workaround can be removed once https://github.com/ballerina-platform/ballerina-library/issues/4870 is addressed.
+
+3. Change the types `Organization_field_id` and `User_field_id` from `int|string?` to `int|string`.
+    * This is done as a post-workaround after using the `--nullable` option to generate the client, as nilable types are not supported for path parameters in Ballerina
+
+4. Add `Accept` header (as `application/json`) to all the operations.
+    * The ZenDesk OpenAPI reference does not include the `Accept` header in the operations, but the server returns an 
+    unsupported media type error if the `Accept` header is not included in the requests.
```
Comment (ayeshLK) [inline docs/spec/sanitations.md:28]: Lets add the OpenAPI command you used to build the connector to this doc.

File: ballerina/client.bal [hunk 2]
```diff
@@ -13,3 +16,5082 @@
...
+            }
+            if config.proxy is http:ProxyConfig {
+                httpClientConfig.proxy = check config.proxy.ensureType(http:ProxyConfig);
+            }
+        }
+        http:Client httpEp = check new (serviceUrl, httpClientConfig);
+        self.clientEp = httpEp;
+        return;
+    }
+    # List assignable groups and agents based on query matched against name
```
Comment (ayeshLK) [inline ballerina/client.bal:57]: Shall we add a newline here ? I think we did this to other connectors as well.

File: ballerina/utils.bal [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    boolean explode = true;
+    # Specifies the custom content type
+    string contentType?;
+    # Specifies the custom headers
+    map<any> headers?;
+};
+
+enum EncodingStyle {
+    DEEPOBJECT, FORM, SPACEDELIMITED, PIPEDELIMITED
+}
```
Comment (Nuvindu) [inline ballerina/utils.bal:38]:
```suggestion
enum EncodingStyle {
    DEEPOBJECT,
    FORM,
    SPACEDELIMITED,
    PIPEDELIMITED
}
```

File: ballerina/utils.bal [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+#
+# + parent - Parent record name
+# + anyRecord - Record to be serialized
+# + explode - Specifies whether arrays and objects should generate separate parameters
+# + return - Serialized record as a string
+isolated function getFormStyleRequest(string parent, record {} anyRecord, boolean explode = true) returns string {
+    string[] recordArray = [];
+    if explode {
+        foreach [string, anydata] [key, value] in anyRecord.entries() {
+            if (value is SimpleBasicType) {
```
Comment (Nuvindu) [inline ballerina/utils.bal:77]:
```suggestion
            if value is SimpleBasicType {
```

Comment (Nuvindu) [reply ballerina/utils.bal:77]: check for other places

File: ballerina/utils.bal [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+# + value - Value to be encoded
+# + return - Encoded string
+isolated function getEncodedUri(anydata value) returns string {
+    string|error encoded = url:encode(value.toString(), "UTF8");
+    if (encoded is string) {
+        return encoded;
+    } else {
+        return value.toString();
+    }
+}
```
Comment (Nuvindu) [inline ballerina/utils.bal:181]:
```suggestion
isolated function getEncodedUri(anydata value) returns string {
    string|error encoded = url:encode(value.toString(), "UTF8");
    if encoded is string {
        return encoded;
    }
    return value.toString();
}
```

File: ballerina/utils.bal [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+// under the License.
+
+import ballerina/url;
+
+type SimpleBasicType string|boolean|int|float|decimal;
+
+# Represents encoding mechanism details.
+type Encoding record {
+    # Defines how multiple values are delimited
+    string style = FORM;
```
Comment (Nuvindu) [inline ballerina/utils.bal:27]:
```suggestion
    EncodingStyle style = FORM;
```

File: ballerina/client.bal (hunk 2 above)
Comment (NipunaRanasinghe) [reply ballerina/client.bal:57]: @ayeshLK IMO we should make minimum(ideally zero) changes to the generated code, and make the generator tools (openAPI tool + formatter) intelligent enough to address those improvements.

Comment (NipunaRanasinghe) [reply ballerina/client.bal:57]: for this particular case, I've already created an issue in the formatter side which they will address with the update 9 release 

https://github.com/ballerina-platform/ballerina-lang/issues/41462

File: ballerina/utils.bal (hunk 3 above)
Comment (NipunaRanasinghe) [reply ballerina/utils.bal:38]: https://github.com/ballerina-platform/module-ballerinax-zendesk/pull/2#discussion_r1533541605

File: ballerina/utils.bal (hunk 4 above)
Comment (NipunaRanasinghe) [reply ballerina/utils.bal:77]: https://github.com/ballerina-platform/module-ballerinax-zendesk/pull/2#discussion_r1533541605

File: ballerina/utils.bal (hunk 5 above)
Comment (NipunaRanasinghe) [reply ballerina/utils.bal:181]: https://github.com/ballerina-platform/module-ballerinax-zendesk/pull/2#discussion_r1533541605

File: ballerina/utils.bal (hunk 6 above)
Comment (NipunaRanasinghe) [reply ballerina/utils.bal:27]: https://github.com/ballerina-platform/module-ballerinax-zendesk/pull/2#discussion_r1533541605

File: docs/spec/sanitations.md (hunk 1 above)
Comment (NipunaRanasinghe) [reply docs/spec/sanitations.md:28]: +1. Updated with https://github.com/ballerina-platform/module-ballerinax-zendesk/pull/2/commits/5e5e1568769de75dc459a73e1039d32409dca422

**Review by Nuvindu — APPROVED** [review]

**Pull request merged**